src/projectviewer/projectviewerdock.cpp

```diff
@@ -4,13 +4,22 @@
 #include <QException>
 #include <QTreeView>
 #include <QWidget>
+#include <QDir>
 
 #include "mainwindow.h"
+#include "projectviewermodel.h"
+#include "projecttreeview.h"
 
 ProjectViewerDock::ProjectViewerDock(QWidget *pParent): QDockWidget(pParent)
 {
     setWindowTitle("Project Viewer");
-    pTreeViewer = new QTreeView;
+    QStringList filters;
+    filters << "*.txt"<<"*.cpp"<<"*.h"<<"*.json"<<"*.c"<<"*.hpp";
```

Comment:
I suppose you will use these file extension in many places. It is better to create constants. It should be one place in code for adding a new extension. It can be a separate helper class or something like this.